### DISCUSSION TITLE
question about camera driver dtbo fiiles in scarthgap

### DISCUSSION THREAD

**Discussion by arrow53** [OE4T/meta-tegra#1748] [General]
in the `nvidia-kernel-oot_36.3.0.bb` recipe I see a bunch of entries for `TEGRA_OOT_CAMERA_DRIVERS`

I'm interested in 
```
    ${KERNEL_MODULE_PACKAGE_PREFIX}kernel-module-nv-imx219 \
```
I don't understand what this is.

Can I make it so this module isn't inserted? 

I'm using extlinux and interested inserting as such
```
FDT /boot/tegra234-p3768-0000+p3767-0005-nv.dtb
FDTOVERLAYS /boot/dtb/overlays/something_imx219.dtbo
```

I basically want to muck with the camera driver but don't want to reflash the image each time I modify the device tree entry.

**Comment by madisongh** [comment]
That package just installs the loadable kernel module under `/lib/modules`.  If you're working on the driver, you should be able to just copy the resulting `.ko` to the right spot in the filesystem and reboot to test.  You might even be able to get away with using `rmmod` and `insmod` to reload it, if you're lucky.

**Reply by arrow53** [reply]
oh, those are kernel modules. Got it. 

I see a `#include "tegra234-p3768-camera-rbpcv2-imx219.dtsi"` in `tegra234-p3768-0000+p3767-0000-dynamic.dts` and I get that the `tegra234-p3768-0000+p3767-0000-dynamic.dts` file is a dtbo that gets loaded as I can see that in the machine file.

If I comment out the imx219.dtsi can I then load it with `FDTOVERLAYS` via extlinux?

Not sure what the best approach is to quickly switch out dtbos

**Reply by madisongh** [reply]
Hmm. That `...dynamic` overlay is part of the plugin manager stuff that gets attached to the UEFI bootloader image and applied by the bootloader before the kernel loads.

If you need to modify it frequently, it would probably be easier to omit the imx219 include from that overlay altogether and directly include it in your main device tree. Then as you make changes, just copy the new main device tree into `/boot` in your rootfs, rather than trying to mess around with the overlay stuff.

Once you've worked out the changes you need, you could always go back to the dynamic overlay approach if that's needed for your normal builds.
